src/include/daos_srv/vos_types.h

```diff
@@ -391,8 +391,10 @@ enum {
 	VOS_IT_UNCOMMITTED = (1 << 8),
 	/** The iterator is for an aggregation operation (EC or VOS) */
 	VOS_IT_FOR_AGG = (1 << 9),
+	/** Checking whether the target is aborted or not. */
+	VOS_IT_FOR_CHECK = (1 << 10),
```

Comment:
It looks to me this VOS_IT_FOR_CHECK/DAOS_INTENT_CHECK is duplicated of VOS_IT_FOR_PURGE/DAOS_INTENT_PURGE?

Reply:
In the long term, it is not impossible to cleanup DAOS_INTENT_XXX, but that needs more time to study every usage carefully and will cause more code changes. If we have to do that, then I prefer to do that after releasing DAOS-2.8.

Reply:
yeah, I think it could be simplified. Looks to me they are the same use case: simply returning every item and associated DTX state to the caller.